File: Longest-Subarray-of-1's-After-Deleting-One-Element.cs
public class Solution {
    public int LongestSubarray(int[] nums) {
        int left = 0 , right = 0 ;
        int maxi = 0 ; 
        int zeroCount = 0 ;
        while(right<nums.Length){
            if(nums[right]==0){
                zeroCount++;
            }

            while(zeroCount>1){
                if(nums[left]==0){
                    zeroCount--;
                }
                left++; 
            }
            maxi = Math.Max(maxi,right-left);
            right++; 
        }

        return maxi ; 
    }
}

File: Longest-Subarray-of-1's-After-Deleting-One-Element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int left = 0 ;
        int max_len = 0 ;
        int last_zero = -1 ;

        for(int right = 0 ;right<nums.size();right++){
            if(nums[right]==0){
                left = last_zero + 1 ;
                last_zero = right;
            }
            max_len = max(max_len,right-left);
        }

        return max_len;
    }
};